src/libraries/System.Private.CoreLib/src/System/IO/StreamWriter.cs

```diff
@@ -613,53 +625,38 @@ public override Task WriteAsync(char value)
             ThrowIfDisposed();
             CheckAsyncTaskInProgress();
 
-            Task task = WriteAsyncInternal(this, value, _charBuffer, _charPos, _charLen, CoreNewLine, _autoFlush, appendNewLine: false);
+            Task task = WriteAsyncInternal(value, appendNewLine: false);
             _asyncWriteTask = task;
 
             return task;
         }
 
-        // We pass in private instance fields of this MarshalByRefObject-derived type as local params
-        // to ensure performant access inside the state machine that corresponds this async method.
-        // Fields that are written to must be assigned at the end of the method *and* before instance invocations.
-        private static async Task WriteAsyncInternal(StreamWriter _this, char value,
-                                                     char[] charBuffer, int charPos, int charLen, char[] coreNewLine,
-                                                     bool autoFlush, bool appendNewLine)
+        private async Task WriteAsyncInternal(char value, bool appendNewLine)
         {
-            if (charPos == charLen)
+            if (_charPos == _charLen)
             {
-                await _this.FlushAsyncInternal(false, false, charBuffer, charPos).ConfigureAwait(false);
-                Debug.Assert(_this._charPos == 0);
-                charPos = 0;
+                await FlushAsyncInternal(flushStream: false, flushEncoder: false).ConfigureAwait(false);
             }
 
-            charBuffer[charPos] = value;
-            charPos++;
+            _charBuffer[_charPos++] = value;
 
             if (appendNewLine)
             {
-                for (int i = 0; i < coreNewLine.Length; i++)   // Expect 2 iterations, no point calling BlockCopy
+                for (int i = 0; i < CoreNewLine.Length; i++) // Generally 1 (\n) or 2 (\r\n) iterations
                 {
-                    if (charPos == charLen)
+                    if (_charPos == _charLen)
                     {
-                        await _this.FlushAsyncInternal(false, false, charBuffer, charPos).ConfigureAwait(false);
-                        Debug.Assert(_this._charPos == 0);
-                        charPos = 0;
+                        await FlushAsyncInternal(flushStream: false, flushEncoder: false).ConfigureAwait(false);
                     }
 
-                    charBuffer[charPos] = coreNewLine[i];
-                    charPos++;
+                    _charBuffer[_charPos++] = CoreNewLine[i];
                 }
             }
 
-            if (autoFlush)
+            if (AutoFlush)
```

Comment:
This was calling non-virtual `_autoFlush` before vs. calling virtual (and potentially overriden) AutoFlush now. Not sure whether it matters.

Reply:
Good catch.  Will fix.